include/aikido/planner/ConfigurationToEndEffectorOffset.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef AIKIDO_PLANNER_CONFIGURATIONTOENDEFFECTOROFFSET_HPP_
+#define AIKIDO_PLANNER_CONFIGURATIONTOENDEFFECTOROFFSET_HPP_
+
+#include <dart/dart.hpp>
+#include "aikido/constraint/Testable.hpp"
+#include "aikido/planner/Problem.hpp"
+#include "aikido/statespace/StateSpace.hpp"
+#include "aikido/trajectory/Interpolated.hpp"
+
+namespace aikido {
+namespace planner {
+
+/// Planning problem to plan to desired end-effector offset while maintaining
+/// the current end-effector orientation.
+class ConfigurationToEndEffectorOffset : public Problem
+{
+public:
+  /// Constructor.
+  ///
+  /// \param[in] stateSpace State space.
+  /// \param[in] endEffectorBodyNode BodyNode to be planned to move to a desired
+  /// offest while maintaining the current orientation.
+  /// \param[in] startState Start state.
+  /// \param[in] direction Unit vector that represents the direction of motion
+  /// [unit vector in the world frame]. The direction is allowed to be zero
+  /// vector.
```

Comment:
Should this docstring mention what direction being zero-vector means?

Reply:
Maybe let's just throw an error if the direction is zero since we shouldn't ever actually want to do that.

Reply:
Yeah.
@jslee02 why is it allowed to be zero vector?